hail/src/main/scala/is/hail/linalg/BlockMatrix.scala

```diff
@@ -1702,6 +1702,7 @@ class BlockMatrixReadRowBlockedRDD(
     val filename = path + "/parts/" + pFile
     val is = localHadoopConfBc.value.value.unsafeReader(filename)
     val in = BlockMatrix.bufferSpec.buildInputBuffer(is)
+    TaskContext.get.addTaskCompletionListener(_ => in.close())
```

Comment:
BURN EVERYTHING IN A FIRE WTF.

WHY ARE WE CAPTURING THE CONTAINING CLASS HERE SCALA. WTF!

```
      98: new           #217                // class is/hail/linalg/BlockMatrixReadRowBlockedRDD$$anonfun$is$hail$linalg$BlockMatrixReadRowBlockedRDD$$readPartFileRow$1
     101: dup
     102: aload_0
     103: aload         9
     105: invokespecial #220                // Method is/hail/linalg/BlockMatrixReadRowBlockedRDD$$anonfun$is$hail$linalg$BlockMatrixReadRowBlockedRDD$$readPartFileRow$1."<init>":(Lis/hail/linalg/BlockMatrixReadRowBlockedRDD;Lis/hail/io/InputBuffer;)V
```